patroni/postgresql.py

```diff
@@ -100,9 +100,13 @@ def __init__(self, config):
         self._postgresql_conf = os.path.join(self._data_dir, config_base_name + '.conf')
         self._postgresql_base_conf_name = config_base_name + '.base.conf'
         self._postgresql_base_conf = os.path.join(self._data_dir, self._postgresql_base_conf_name)
+        self._postgresql_custom_conf = config.get('custom_conf')
         self._recovery_conf = os.path.join(self._data_dir, 'recovery.conf')
-        self._configuration_to_save = (self._postgresql_conf, self._postgresql_base_conf,
-                                       os.path.join(self._data_dir, 'pg_hba.conf'))
+        self._configuration_to_save = [self._postgresql_conf]
+        if not self._postgresql_custom_conf:
```

Comment:
`postgresql.custom_conf` and `postgresql.parameters.hba_conf` can be set, updated and removed from the configuration in a run-time either by updating of `patroni.yaml` config file manually and sending SIGHUP or by changing value of the cluster `config` in DCS.

`_configuration_to_save` if used to make a copies of the configuration files to be backed up by wal-e (it skips all *.conf files)
In case of external configuration files I think it is still make sense to back them up by copying `custom_conf` and `hba_file` into `data_dir` with the names "postgresql.base.conf.backup" and "pg_hba,conf.backup" accordingly.
 
Such functionality requires some changes in a `save_configuration_files` and `restore_configuration_files` methods and we will need to change the way we keep list of configuration files from `list` or `tuple` to `dict`.

`Postgresql.reload_config` method should update value `_configuration_to_save` if the `custom_conf` or `hba_file` were changed.